chroma-agent/chroma_agent/action_plugins/manage_node.py

```diff
@@ -75,4 +76,18 @@ def initialise_block_device_drivers():
 
     return agent_result_ok
 
+
+# When the agent is stopped we want to allow block devices to do any termination that they might need, this function
+# may also be called by the manager.
+@agent_daemon_teardown_function()
+def terminate_block_device_drivers():
+    console_log.info("Terminating drivers for block device types")
+    for cls in util.all_subclasses(BlockDevice):
+        error = cls.terminate_driver(config.profile_managed)
```

Comment:
removed, it is unused because we need to use zpool transition locks whether in monitored or managed mode